### PR TITLE
Add SlotsView Wrapper to View

### LINKED ISSUE #1743: Add `__slots__` to `arcade.View`
### What should be added/changed?

As talked about in Discord, a `__slots__` attribute could be added to `arcade.View`.

### What would it help with?

This will make it easier to specify which attributes belong to a class and should also improve memory performance.

### PR DESCRIPTION
**Pull request by gran4**
Fix https://github.com/pythonarcade/arcade/issues/1743. Basically View now inherits from slots view.

### CODE REVIEW

File: arcade/application.py [hunk 1]
```diff
@@ -934,7 +953,12 @@ class View:
     """
     Support different views/screens in a window.
     """
-
+    __slots__ = (
+        "window",
+        "key",
+        "_section_manager",
+        "section_manager",
```
Comment (gran4) [inline arcade/application.py]: @einarf , it seems that some variables are not created in the class, so it is having issues? What should I do? I can't find anythin on it.

Comment (gran4) [reply arcade/application.py]: Like section_manager here `conflicts with class variable`

Comment (einarf) [reply arcade/application.py]: Slots is only for the data members. Not for properties or methods. The actual data memers is what we create instance data from.

I would skip doing slots for the Window itself. Just focus on the View. The main reason for useing slots of to save some memory per instance (no internal member dict).

**Review by einarf — CHANGES_REQUESTED** [review]

**Comment by gran4** [comment]
Thats what I figured might be it.

**Comment by gran4** [comment]
What do I do about `AttributeError: 'View' object attribute 'on_hide_view' is read-only`?
It is a function

**Comment by pushfoo** [comment]
As I commented on the original issue and discord, I'm concerned by the following:
1. The effects this might have on mix-ins
2.  Running into memory issues from a large number of views suggests potential underlying issues with a project's design. 

@einarf Would the following be an acceptable middle ground?

1. Define a `ViewProtocol` type mirroring the API of the current `View` class
3. Keep the current view class as-is
4. Optionally, add a performance-oriented `SlotsView` base class

**Comment by gran4** [comment]
I think, you should try to use slots, but I think I understand why someone would want to add values to View.

EDIT: so not 2? Do 3?

EDIT pt.2: Where is this going @pushfoo ???

**Comment by gran4** [comment]
The wrapper will not work, so we need a different solution.

P.S: @pvcraven I believe this is the PR you were looking for about slots in View. I do not know if this will help at all, lol.

**Comment by einarf** [comment]
I think this was not a good idea in the first place. It's easier to see the issues once the change have been made. I think my comments on discord was ill-advised.

@pushfoo 's idea about making a protocol is not a bad one I think. Could be that we need different view types in the future? I think we'll just close this is for now. It's a bit too radical at this point in time and we really need to get this 3.0 release out soon.

Thanks for giving it an attempt at least! It was still useful in some way.

**Pull request closed without merging**